File: src/dataset/python/pig-latin_exercism-ThomasZumsteg/openAI/ChatGPT4_pig_latin_v4.py
import re

def translate(phrase):
    """translate converts a phrase to piglatin"""
    return ' '.join(translate_word(word) for word in phrase.split())

CASES = [
    re.compile(r'(.*?[^q])(u.*)'),  # q case
    re.compile(r'(^y[^aeiou].*)'),  # y case
    re.compile(r'(.*?)([aeoi].*)')  # default case
]

def translate_word(word):
    """translate_word converts a word to piglatin"""
    for case in CASES:
        match = case.match(word)
        if match:
            prefix, suffix = match.groups(default="")
            if prefix + "ay" != word:
                return suffix + prefix + "ay"
    return word + "ay"

File: src/dataset/javascript/pig-latin_exercism-ThomasZumsteg/openAI/ChatGPT4_pig-latin_v4.js
function translate(phrase) {
	return phrase.split(' ').map(pigize).join(' ');
}

function pigize(word) {
	const regex = /^([^aeioy]*qu)(.*)$|^(.*?)([aeiouy].*)$/;
	const matches = word.match(regex);
	return matches ? (matches[2] || matches[4]) + (matches[1] || matches[3]) + "ay" : word;
}

module.exports = { translate };